pkg/cli/image/manifest/manifest.go

```diff
@@ -388,7 +388,7 @@ func ManifestToImageConfig(ctx context.Context, srcManifest distribution.Manifes
 		return base, layers, nil
 
 	default:
-		return nil, nil, fmt.Errorf("unknown image manifest of type %T from %s", srcManifest, location)
+		return nil, nil, fmt.Errorf("use --keep-manifest-list option for found image manifest type %T from %s", srcManifest, location)
```

Comment:
Not quite, I'd suggest adding a new case matching only ManifestList types (check to use all or an interface that will cover all) and only suggest `--keep-manifest-list` for those, and keep the default as it was before. This might be confusing as it stands.

Reply:
I think, `*manifestlist.DeserializedManifestList` is the thing you suggested